build-support/bin/pre-commit.sh

```diff
@@ -51,36 +51,37 @@ printf "%s\n" "${ADDED_FILES[@]}" \
 echo "* Checking for banned imports"
 ./build-support/bin/check_banned_imports.sh
 
-if git diff master --name-only | grep '\.rs$' > /dev/null; then
-  echo "* Checking formatting of rust files" && ./build-support/bin/check_rust_formatting.sh || exit 1
-  # Clippy happens on a different shard because of separate caching concerns.
-  if [[ "${RUNNING_VIA_TRAVIS_CI_SCRIPT}" != "1" ]]; then
-    echo "* Running cargo clippy" && ./build-support/bin/check_clippy.sh || exit 1
-  fi
-  echo "* Checking rust target headers" && build-support/bin/check_rust_target_headers.sh || exit 1
-fi
-
 echo "* Checking for bad shell patterns" && ./build-support/bin/check_shell.sh || exit 1
 
-$(git rev-parse --verify master > /dev/null 2>&1)
-if [[ $? -eq 0 ]]; then
+# When travis builds a tag, it does so in a shallow clone without master fetched, which
+# fails in pants changed.
+master_present=$(git rev-parse --verify master > /dev/null 2>&1 || echo "Nope!")
```

Comment:
Could the string "Nope!" or the variable name `master_present` be made more descriptive?

Reply:
Also, I'm aware the docstring is pulled from lower in the file, but it would be good to say why `./pants changed` fails -- it's specifically because we're calculating the changes since master in the command line, and then run other `git diff master` commands.

Reply:
I would actually prefer that the branch name be made its own variable, and we remove the string entirely:
```bash
# The branch to calculate changed targets and files against.
BASE_BRANCH="${BASE_BRANCH:-master}"
# Test if this git repo contains the specified base branch. When travis builds a tag, it does so in
# a shallow clone without master fetched, so we currently can't calculate changes against it.
if git rev-parse --verify "$BASE_BRANCH" &>/dev/null; then
  echo "* Checking imports" && ./build-support/bin/isort.sh || \
  # ...
```

Reply:
> if git rev-parse --verify "$BASE_BRANCH" &>/dev/null; then

The issue with this line would be that the new `set -e` in this file would cause the script to exit here.

Reply:
Ah... my bash is weak. It looks like that properly handles the exit. Will update.

Reply:
> Not all instance of `master` are in this file, so I don't know if that would be effective without additional refactoring.

Didn't think hard enough about that one! Would love to parameterize this in the future for cleanliness but something to address in full later.